components/ActionSheet/src/private/MDCActionSheetListViewController.m

```diff
@@ -0,0 +1,157 @@
+  return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return self.actions.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  MDCActionSheetItemView *cell =
+  [[MDCActionSheetItemView alloc] initWithAction:self.actions[indexPath.row]
+                                 reuseIdentifier:kReuseIdentifier];
+  cell.backgroundColor = self.backgroundColor;
+  cell.actionsFont = _actionsFont;
+  return cell;
+}
+
+#pragma mark - Dynamic type
+
++ (UIFont *)actionsFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleSubheadline];
+  }
+  return [MDCTypography subheadFont];
+}
+
+- (BOOL)mdc_adjustFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+  [self updateFonts];
+}
+
+
+- (void)updateFonts {
+  UIFont *finalFont = _actionsFont ?: [[self class] actionsFontDefault];
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    finalFont =
+        [finalFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleSubheadline
+                              scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+  }
+  _actionsFont = finalFont;
+  [self.tableView reloadData];
+  [self.view setNeedsLayout];
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  self.view.backgroundColor = backgroundColor;
+  self.tableView.backgroundColor = backgroundColor;
+  [self.view setNeedsLayout];
+}
+
+- (UIColor *)backgroundColor {
+  return self.view.backgroundColor;
+}
+
+#pragma mark - Setters / Getters
+
+- (void)setActionsFont:(UIFont *)actionsFont {
+  _actionsFont = actionsFont;
+  [self updateFonts];
+}
+
+- (CGFloat)tableHeightForWidth:(CGFloat)width {
+  CGFloat height = 0.f;
+  UILabel *mockLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+  mockLabel.font = _actionsFont;
+  for (MDCActionSheetAction *action in _actions) {
+    CGFloat leadingPadding = (action.image == nil) ? 16.f : 72.f;
+    NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+    paragraphStyle.hyphenationFactor = 1.f;
+    NSDictionary<NSAttributedStringKey, id> *attributes =
+        @{ NSParagraphStyleAttributeName : paragraphStyle };
+    NSMutableAttributedString *attributedString =
+    [[NSMutableAttributedString alloc] initWithString:action.title attributes:attributes];
+    mockLabel.attributedText = attributedString;
+    CGSize labelSize = CGRectInfinite.size;
+    labelSize.width = width - kActionItemTrailingPadding - leadingPadding;
+
+    CGFloat labelHeight = [mockLabel sizeThatFits:labelSize].height;
+    height = height + labelHeight + (2 * kActionItemLabelPadding);
+  }
+  return height;
```

Comment:
I wager you can probably rely on UITableView's contentSize.height calculations so that you don't have to calculate the height yourself.

Reply:
I may be layout out my table incorrectly but I couldn't get contentSize.height to respond to dynamic type. The contentSize.height is always the same weather in the smallest contentSize or largest.